src/bsoncxx/CMakeLists.txt

```diff
@@ -160,8 +160,8 @@ install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
     FILES_MATCHING
       PATTERN "*.h"
       PATTERN "*.hpp"
-      REGEX ".+/private" EXCLUDE
-      REGEX ".+/test" EXCLUDE
+      REGEX "${CMAKE_CURRENT_SOURCE_DIR}/.*private" EXCLUDE
```

Comment:
Will `${CMAKE_CURRENT_SOURCE_DIR}` will be interpreted as part of a regex?  If so, that doesn't seem quite correct to me, in the case where the filename contains any special characters.  Note also that you removed the slash preceding `private`, which I think should be restored.

I read the relevant CMake docs and can't think of a bulletproof counter-proposal, though.  Can you use `\Q` and `\E` in these regular expressions?  That won't work if the filename includes `\E`, but it's probably good enough for our purposes.